crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -27,22 +29,55 @@ impl std::ops::Deref for Direction3d {
 }
 
 /// A sphere primitive
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq)]
 pub struct Sphere {
     /// The radius of the sphere
     pub radius: f32,
 }
 impl Primitive3d for Sphere {}
 
+impl Sphere {
+    /// Get the diameter of the sphere
+    pub fn diameter(&self) -> f32 {
+        2.0 * self.radius
+    }
+
+    /// Get the surface area of the sphere
+    pub fn area(&self) -> f32 {
+        4.0 * PI * self.radius.powi(2)
+    }
+
+    /// Get the volume of the sphere
+    pub fn volume(&self) -> f32 {
+        4.0 * FRAC_PI_3 * self.radius.powi(2) * self.radius
```

Comment:
Shouldn't this be `self.radius.powi(3)` if we are using `powi`?

Reply:
`r² × r` is the same as `r³`, so it makes no numerical difference

Reply:
I know, it's not a numerical concern, more of a code cleanliness one, either you're writing your product explicitly, or you can use an exponent, no reason I see to write it with both a product and an exponent.

So I'm suggesting either `r * r * r` or `r³`, no need to mix.